### PR TITLE
docs: add template reference docs for Python

### PR DESCRIPTION
**Pull request by lance**
Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

File: docs/guides/python.md
Comment (abrennan89) [inline docs/guides/python.md:142]: Can you add an example of what that txt file would look like with a python dependency added just for completeness? Or a command that would add it, do they add it manually, etc?

Comment (zroubalik) [reply docs/guides/python.md:142]: `requirements.txt` could be considered as a standard Python way of describing depenedencies, so every Python dev knows how to manage this file. IMHO I don't think it is neccessary to desribe it for function developers.

**Review by zroubalik — APPROVED** [review]
LGTM

File: docs/guides/python.md
Comment (lance) [reply docs/guides/python.md:142]: I agree with @zroubalik this will be fundamental knowledge for any Python dev.

Let's hold this PR until @matejvasek's [PR](https://github.com/boson-project/parliament/pull/9) lands since that will affect the docs for `context`.

File: docs/guides/python.md
Comment (matejvasek) [inline docs/guides/python.md:118]: So user cannot return just CloudEvent, meaning user has to transform CloudEvent to HTTP response by himself?

Comment (lance) [reply docs/guides/python.md:118]: Good point. That is correct for now.

Comment (lance) [reply docs/guides/python.md:118]: Working on adding a `@event` decorator to parliament.

Comment (lance) [reply docs/guides/python.md:118]: https://github.com/boson-project/parliament/pull/17

File: docs/guides/python.md
Comment (zroubalik) [inline docs/guides/python.md:21]:
```suggestion
❯ kn func create -l python fn
Project path: /home/developer/projects/fn
Function name: fn
Runtime: python
Trigger: http

❯ tree
fn
├── func.py
├── func.yaml
├── requirements.txt
└── test_func.py
```

Making this consistent with other Guides.

File: docs/guides/python.md
Comment (zroubalik) [inline docs/guides/python.md:101]:
```suggestion
When an incoming request is received, your function will be invoked with a `Context` object as the first parameter. This object is simply a Python `dict` with two keys, `request` and
```

**Comment by lance** [comment]
@zroubalik @matejvasek @abrennan89 - ok I think this is ready for a final review before landing. Thanks. :)

File: docs/guides/python.md
Comment (zroubalik) [inline docs/guides/python.md:9]: The rest of the Guide (as well as the others) shows usage of `func` not `kn func`. Would be nice to make it consistent.
```suggestion
❯ func create -l python fn
```

File: docs/guides/python.md
Comment (zroubalik) [inline docs/guides/python.md:17]:
```suggestion
❯ tree
fn
```
This is what we are using in the other guides, but feel free to not use this suggestion :)

**Review by zroubalik — APPROVED** [review]
Looking good, minor nitpicking, feel free to ignore those suggestions :)

File: docs/guides/python.md
Comment (zroubalik) [inline docs/guides/python.md:1]:
```suggestion
# Python Function Developer's Guide
```

**Comment by lance** [comment]
@zroubalik I appreciate the keen eye - thanks for the feedback!

**Pull request merged**
